test/unit/Assessment/withdrawRewards.js

```diff
@@ -183,4 +183,116 @@ describe('withdrawRewards', function () {
       .to.emit(assessment, 'RewardWithdrawn')
       .withArgs(staker.address, staker.address, totalRewardInNXM);
   });
+
+  it('reverts if system is paused', async function () {
+    const { assessment, master, individualClaims } = this.contracts;
+    const [staker] = this.accounts.members;
+
+    const { minVotingPeriodInDays, payoutCooldownInDays } = await assessment.config();
+    await assessment.connect(staker).stake(parseEther('10'));
+
+    await individualClaims.connect(staker).submitClaim(0, 0, parseEther('100'), '');
+    await assessment.connect(staker).castVotes([0], [true], 0);
+    const { timestamp } = await ethers.provider.getBlock('latest');
+    await setTime(timestamp + daysToSeconds(minVotingPeriodInDays + payoutCooldownInDays));
+
+    await master.setEmergencyPause(true);
+
+    await expect(assessment.connect(staker).withdrawRewards(staker.address, 0)).to.be.revertedWith('System is paused');
+  });
+
+  it('reverts if staker is not a member', async function () {
+    const { assessment, individualClaims } = this.contracts;
+    const [staker] = this.accounts.members;
+    const [nonMember] = this.accounts.nonMembers;
+
+    const { minVotingPeriodInDays, payoutCooldownInDays } = await assessment.config();
```

Comment:
is this code block repeated more than 2 times?

we could use a function for it then.

I can see the other cases may have different setups so that can stay the same.

Reply:
Thanks for the suggestion! Created the helper methods `finalizePoll` and `generateRewards` to reuse them in the tests to avoid duplicated code